src/core/searchUtils.js

```diff
@@ -92,6 +92,22 @@ export function loadSearchResultsIfNeeded({ store: { dispatch, getState }, locat
     ...location.query,
     clientApp: state.api.clientApp,
   });
+
+  if (!isLoaded({ state: state.search, page, filters })) {
+    return performSearch({ dispatch, page, api: state.api, auth: state.auth, filters });
+  }
+  return true;
+}
+
+export function loadCategoryResultsIfNeeded({ store: { dispatch, getState }, location, params }) {
+  const page = parsePage(location.query.page);
+  const state = getState();
+  const filters = {
+    addonType: params.addonType,
+    category: params.slug,
+    clientApp: params.application,
+  };
+
   if (!isLoaded({ state: state.search, page, filters })) {
```

Comment:
Why do the search results need to be cached? I think it should make an API request every time. Otherwise, the single page app will go stale -- the only way a user can obtain newly added results is to refresh the page. The browser will already cache the API call as needed so I don't really see the value in adding a new caching layer on top of that.